### PR TITLE
feat(select): checkbox in select field options

### PR DESCRIPTION
**Pull request by pnzau**
Add checkboxes to select, so that clicking on an option can be done through a checkbox

CCIBI-4877

### CODE REVIEW

File: src/Select/Select.component.tsx [hunk 1]
```diff
@@ -184,6 +185,13 @@ export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    * @default false
    **/
   highlightFilled?: boolean;
+
+  /**
+   * Placeholder text
```
Comment (andrey-liventsev-ontarget) [inline src/Select/Select.component.tsx:190]: Please describe the new option. It is not clear now.

File: src/Select/Select.component.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import Select, { Creatable as CreatableSelect } from 'react-select';
 import styled, { ThemeProvider } from 'styled-components';
 import { Themes } from '../themes';
 import uuid from 'uuid';
+import { SelectCheckboxProps } from './SelecCheckbox.component';
```
Comment (andrey-liventsev-ontarget) [inline src/Select/Select.component.tsx:7]: Typo in file name

File: src/Select/Select.stories.tsx [hunk 3]
```diff
@@ -163,111 +147,28 @@ class MultiSelectCheckbox extends React.Component<Props, State> {
...
         <Select
           creatable={boolean('creatable', true)}
-          isMulti
-          components={{
-            Option: this.Option,
-            MultiValue: this.MultiValue,
-            MenuList: this.MenuList,
-          }}
+          isMulti={boolean('isMulti', true)}
+          optionType={'checkbox'}
```
Comment (andrey-liventsev-ontarget) [inline src/Select/Select.stories.tsx:163]: Please provide a knob to manage optionType value. In same way as selectSize

**Review by andrey-liventsev-ontarget — CHANGES_REQUESTED** [review]

File: src/Select/Select.stories.tsx (hunk 3 above)
Comment (pnzau) [reply src/Select/Select.stories.tsx:163]: will do, thanks

Comment (pnzau) [reply src/Select/Select.stories.tsx]: fixed!

File: src/Select/Select.component.tsx (hunk 2 above)
Comment (pnzau) [reply src/Select/Select.component.tsx]: fixed

File: src/Select/Select.component.tsx (hunk 1 above)
Comment (pnzau) [reply src/Select/Select.component.tsx]: fixed!

**Review by andrey-liventsev-ontarget — APPROVED** [review]

**Pull request merged**